test/e2e/page-objects/pages/dialog.ts

```diff
@@ -0,0 +1,47 @@
+import { Driver } from '../../webdriver/driver';
+import { RawLocator } from '../common';
+
+class Dialog {
+  private driver: Driver;
+
+  private setApprovalForAllTitleElement: RawLocator;
+
+  private setApprovalForAllSubHeadingElement: RawLocator;
+
+  private scrollToBottomButton: RawLocator;
+
+  private footerConfirmButton: RawLocator;
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+
+    this.setApprovalForAllTitleElement = {
+      css: 'h2',
+      text: 'Withdrawal request',
+    };
+    this.setApprovalForAllSubHeadingElement = {
+      css: 'p',
+      text: 'This site wants permission to withdraw your NFTs',
+    };
+    this.scrollToBottomButton = '.confirm-scroll-to-bottom__button';
+    this.footerConfirmButton = '[data-testid="confirm-footer-button"]';
+  }
+
+  public async check_setApprovalForAllTitle() {
+    await this.driver.waitForSelector(this.setApprovalForAllTitleElement);
+  }
+
+  public async check_setApprovalForAllSubHeading() {
```

Comment:
As an example, this feels suitable in a `ApprovalConfirmation` or `ApprovalTransactionConfirmation`.